gricleaner.py

```diff
@@ -76,22 +76,31 @@ def get_digest(self, repo, tag):
         response = requests.head(self.registry + path, headers=headers, verify=self.requests_verify)
         return response.headers["Docker-Content-Digest"]
 
-    def delete_image(self, repo, tag):
+    def delete_image(self, repo, tag, purge):
         """Delete image by tag from registry"""
-        url = "/v2/{}/manifests/{}".format(repo, self.get_digest(repo, tag))
-        logging.debug("Delete URL: {}{}".format(self.registry, url))
-        if self.dry_run:
-            logging.warning("~ Dry Run!")
-        else:
-            headers = {
-                "Authorization": "Bearer " + self.get_bearer("repository:" + repo)
-            }
-            response = requests.delete(self.registry + url, headers=headers, verify=self.requests_verify)
-            if response.status_code == 202:
-                logging.info("+ OK")
-            else:
-                logging.error(response.text)
-
+        counter = 0
+        while counter < 2:
```

Comment:
yuk. perhaps add another method to the class `delete_with_purge`, instead of making original `delete` method such ugliness?